text/0118-search-api.md

```diff
@@ -426,9 +431,146 @@ Sets the maximum number of documents to be returned for the search query.
+#### 3.1.7. `page`
+
+- Type: Integer
+- Required: False
+- Default: `null`
+
+Sets the specific results page.
+
+By default, page is `null`, or `1` if `hitsPerPage` is provided. 
+The first page has a value of `1`, the second `2`, etc...
+
+When providing `page` or `hitsPerPage` in the query parameters, the `page selection` system is enabled, which impacts the returned fields and the performance. See explanation on the [`page selection`](#3181-navigating-search-results-by-page-selection).
+
+If in addition to either `page` and/or `hitsPerPage`, `limit` and/or `offset` are provided as well, `limit` and `offset` are ignored. See [explaination](#3181-navigating-search-results-by-page-selection).
+
+- 🔴 Sending a value with a different type than `Integer` for `page` returns a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
+
+#### 3.1.8. `hitsPerPage`
+
+- Type: Integer
+- Required: False
+- Default: `null`
+
+Sets the number of results returned for a query.
+
+By default, `hitsPerPage` is `null`, or `20` if `page` is provided.
+
+When providing `page` or `hitsPerPage` in the query parameters, the `page selection` system is enabled, which impacts the returned fields and the performance. See explanation on the [`page selection`](#3181-navigating-search-results-by-page-selection).
+
+If in addition to either `page` and/or `hitsPerPage`, `limit` and/or `offset` are provided as well, `limit` and `offset` are ignored. See [explaination](#3181-navigating-search-results-by-page-selection).
+
+- 🔴 Sending a value with a different type than `Integer` for `hitsPerPage` returns a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
+
+#### 3.1.8.1. Navigating search results by page selection
+
+By default, `limit` and `offset` are used for pagination. That pagination system, while performant, does not provide the required information to create a correct page selection. Upon using `limit`/`offset`, `estimatedTotalHits` is returned which provides a rough estimation of how many hits may be candidates for a given request. See [`limit`/`offset` usage](#31811-limitoffset-usage) for further explaination.
+
+The `page selection` system provides an alternative that tackles the above-mentioned issue when the user needs reliable information for its pagination. For example, when creating a pagination with numbers `<< < 1, 2, 3, ...14 > >>`. Nonetheless, it is less performant as the engine needs to compute the `totalHits` exhaustively. 
+With this page selection system, it is possible to jump from one page to another using the `page` parameter and decide how many hits should be present in a page with `hitsPerPage`. See [`page`/`hitsPerPage` usage](#31812-pagehitsperpage-usage) for further explaination.
+
+##### 3.1.8.1.1. Limit/offset usage
+
+When either `limit` or `offset` is specified or when neither `limit`, `offset`, `page` and `hitsPerPage` are specific, the response object contains related fields:
```

Comment:
```suggestion
When either `limit` or `offset` is specified or when neither `limit`, `offset`, `page` and `hitsPerPage` are specific, the response object contains the related fields:
```

Reply:
It's seems contradictory with the statement bieng made in `##### 3.1.8.1.2 page/hitsPerPage usage`

👇

As soon as either `page` or `hitsPerPage` is used as a query parameter, in the response object, `limit`, `offset`, and `estimatedTotalHits` are removed and new fields are returned:

Reply:
How is it contradictory? 

This is what it says to enable limit/offset pagination: 
- `limit` or `offset`
- none of the 4th

My bad if I don't see it

Reply:
My bad you are right!